src/linter.js

```diff
@@ -383,13 +386,17 @@ export default class Linter {
         return scanner.scan();
       })
       // messages should be a list of raw message data objects.
-      .then((messages) => {
-        for (let message of messages) {
+      .then(({linterMessages, scannedFiles}) => {
+        for (let message of linterMessages) {
           if (typeof message.type === 'undefined') {
             throw new Error('message.type must be defined');
           }
           this.collector._addMessage(message.type, message);
         }
+
+        for (let filename of scannedFiles) {
```

Comment:
Should deal with this everywhere and add the necessary lint rules to prefer const at some point since using const here is a little counter intuitive. See also http://www.ecma-international.org/ecma-262/6.0/index.html#sec-for-in-and-for-of-statements-static-semantics-boundnames